src/Config.js

```diff
@@ -78,6 +78,58 @@ module.exports = class extends ChainedMap {
     }));
   }
 
+  toString(verbose = false) {
+    const stringify = require('javascript-stringify');
+    const pluginRE = /(?:function|class) (\w+Plugin)/;
+
+    const config = this.toConfig();
+
+    return stringify(config, (value, indent, stringify) => {
+      // shorten long functions
+      if (!verbose && typeof value === 'function' && value.toString().length > 100) {
+        return `function () { /* omitted long function */ }`;
+      }
+
+      // improve plugin output
+      if (value && value.__pluginName) {
+        let match = (
+          value.constructor &&
+          value.constructor.toString().match(pluginRE)
+        );
+        // special case for copy-webpack-plugin which uses a non-standard constructor
+        if (value.__pluginName === 'copy') {
+          match = [null, `CopyWebpackPlugin`];
+        }
+        const name = match[1];
+        const prefix = `/* config.plugin('${value.__pluginName}') */\n`;
```

Comment:
Sorry forgot one last thing: I think it might be good to allow customising the identifier used for the config object. eg Neutrino would need these to say `/* neutrino.config.plugin(...) */`.